Refactor to use the new multiple controller system

DrSmugleaf opened the pull request:
Requires https://github.com/space-wizards/RobustToolbox/pull/1189

ProfanedBane commented:
Throwing an item while inside a locker shoves the locker downward slightly, throwing again with an empty hand generates a server exception.

`[ERRO] runtime: Caught exception in "GameLoop Tick"
System.NullReferenceException: Object reference not set to an instance of an object.
   at Robust.Server.GameObjects.Components.Container.BaseContainer.CanInsert(IEntity toinsert) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Server\GameObjects\Components\Container\Container.cs:line 159
   at Content.Server.GameObjects.HandsComponent.CanDrop(String slot) in D:\Bob\Downloads\space-station-14\Content.Server\GameObjects\Components\GUI\ServerHandsComponent.cs:line 368
   at Content.Server.Interfaces.GameObjects.Components.Interaction.HandsSystem.HandleThrowItem(ICommonSession session, GridCoordinates coords, EntityUid uid) in D:\Bob\Downloads\space-station-14\Content.Server\GameObjects\EntitySystems\HandsSystem.cs:line 148
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.<>c__DisplayClass2_0.<.ctor>b__0(PointerInputCmdArgs& args) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:line 92
   at Robust.Shared.Input.Binding.PointerInputCmdHandler.HandleCmdMessage(ICommonSession session, InputCmdMessage message) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\Input\Binding\InputCmdHandler.cs:line 114
   at Robust.Server.GameObjects.EntitySystems.InputSystem.InputMessageHandler(InputCmdMessage message, EntitySessionEventArgs eventArgs) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Server\GameObjects\EntitySystems\InputSystem.cs:line 63
   at Robust.Shared.GameObjects.EventBusExt.HandlerWrapper`1.Invoke(EntitySessionMessage`1 msg) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EventBusExt.cs:line 24
   at Robust.Shared.GameObjects.EntityEventBus.<>c__DisplayClass7_0`1.<SubscribeEvent>b__0(Object ev) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.cs:line 151
   at Robust.Shared.GameObjects.EntityEventBus.ProcessSingleEvent(EventSource source, Object eventArgs) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityEventBus.cs:line 285
   at Robust.Server.GameObjects.ServerEntityNetworkManager.DispatchEntityNetworkMessage(MsgEntity message) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityNetworkManager.cs:line 161
   at Robust.Server.GameObjects.ServerEntityNetworkManager.Update() in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityNetworkManager.cs:line 56
   at Robust.Shared.GameObjects.EntityManager.Update(Single frameTime) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\GameObjects\EntityManager.cs:line 87
   at Robust.Server.GameObjects.ServerEntityManager.Update(Single frameTime) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Server\GameObjects\ServerEntityManager.cs:line 962
   at Robust.Server.BaseServer.Update(FrameEventArgs frameEventArgs) in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Server\BaseServer.cs:line 537
   at Robust.Shared.Timing.GameLoop.Run() in D:\Bob\Downloads\space-station-14\RobustToolbox\Robust.Shared\Timing\GameLoop.cs:line 195`

ProfanedBane commented:
I can't seem to throw through tables and other things you'd expect to be able to throw over.

DrSmugleaf commented:
> Throwing an item while inside a locker shoves the locker downward slightly, throwing again with an empty hand generates a server exception.
> 
> https://pastebin.com/rusF3SVU

Moved to https://github.com/space-wizards/space-station-14/issues/1465 and https://github.com/space-wizards/space-station-14/issues/1466

> I can't seem to throw through tables and other things you'd expect to be able to throw over.

Moved to https://github.com/space-wizards/space-station-14/issues/1467
